Revert to connect in the constructor

mwileczka opened the pull request:
Upgraded to russh version that properly closes channels, permitting unlimited sequential executes over a single connection. The aforementioned limit on open channels still applies and restricts the number of simultaneous executes.

mwileczka committed:
Revert to connect in the constructor

Upgraded to russh version that properly closes channels, permitting unlimited sequential executes over a single connection. The aforementioned limit on open channels still applies and restricts the number of simultaneous executes.

mwileczka commented:
@mTsBucy1 as soon as this is merged, I'll create a PR for key authentication and server key checking

mTsBucy1 commented on README.md:
```diff
@@ -1,18 +1,17 @@
+for rust with the tokio runtime. Powered by russh.
+
 [Docs.rs](https://docs.rs/async-ssh2-tokio/latest/async_ssh2_tokio/),
 [Crates.io](https://crates.io/crates/async-ssh2-tokio)
 
-This library is an asynchronous and easy-to-use high level ssh client library
-for rust with the tokio runtime. Powered by the rust ssh implementation
-[russh](https://github.com/warp-tech/russh).
-
 
 ## Features
-* Connect to a SSH Host via IP and password
+* Connect to SSH Host via IP and password.
 * Execute commands on the remote host
-* Get the stdout and exit code of the command
```
I think this got mingled in the revert. Please exclude that from the PR.

mTsBucy1 commented on README.md:
```diff
@@ -54,4 +53,4 @@ or set the following environment variables for a working ssh host:
 * `ASYNC_SSH2_TEST_HOST_PW`: The corresponding password. Since this is plain text, creating a new unpriviledged user is recommended.
 
-Note: The doc tests do not use these variables and are therefore not run, but only compiled.
+Note: The doc tests do not use these variables and are therefor not run, but only compiled.
```
Same here

mTsBucy1 commented on src/client.rs:
```diff
@@ -186,87 +181,91 @@ pub struct CommandExecutedResult {
 #[derive(Clone)]
 struct ClientHandler;
 
+impl ClientHandler {
+    fn new() -> Self {
+        Self {}
+    }
+}
```
Unrelated to the connection code, I removed this since you could just use the struct constructor. I find this to be easier to read and it is not exposed to in the api, thus it doesn't matter if we remove it. I suggest removing it again.

mTsBucy1 commented on src/client.rs:
```diff
@@ -186,87 +181,91 @@ pub struct CommandExecutedResult {
-    pub actual_address: SocketAddr,
+}
+
+impl CommandExecutedResult {
+    fn new(output: String, exit_status: u32) -> Self {
+        Self {
+            output,
+            exit_status,
+        }
+    }
 }
```
Same comment as for ClientHandler

mTsBucy1 commented on src/lib.rs:
```diff
@@ -1,13 +1,11 @@
-//! This library is an asynchronous and easy-to-use high level ssh client library
-//! for rust with the tokio runtime. Powered by the rust ssh implementation
-//! [russh](https://github.com/warp-tech/russh), a fork of thrussh.
+//! This library, async-ssh2-tokio, is a asynchronous and super-easy-to-use high level ssh client library for rust.
+//! Powered by russh, a fork of thrussh.
 //!
 //! The heart of this library is [`client::Client`]. Use this for connection, authentification and execution.
 //!
 //! # Features
-//! * Connect to a SSH Host via IP and password
+//! * Connect to SSH Host via IP and password.
 //! * Execute commands on the remote host
-//! * Get the stdout and exit code of the command
```
This too, probably wasn't meant to be changed and just go mingled in the revert.

mTsBucy1 commented on README.md:
```diff
@@ -28,11 +27,11 @@ use async_ssh2_tokio::client::{Client, AuthMethod};
     // Only ip and password based authentification is implemented.
     // If you need key based authentification, create github issue or contribute.
-    let mut client = Client::new(
+    let mut client = Client::connect(
```
This is just a note that this is a breaking change. Therefore this would need to release as 0.5.0.

mTsBucy1 reviewed (changes requested):
Thanks for your work updating the russh version. However, there are some clear revert oversights included that would be best reworked.

Also, I suggest you create the PR (after cleaning up the commit and bumping the version) in the original repo and not my fork.

mwileczka commented:
@mTsBucy1 I had just reverted the entire commit, updated the russh version, and anything required for it to compile. The unit tests were passing on my machine before I submitted PR. I didn't try to go through all of the changes and pick out what to keep/not keep. I can go back and re/undo the updates you commented on and check for others.

I read some of you comments on other places about the original library with its owner, but I wasn't sure where there ended up in terms of you merging, updating, etc with the original.

Are you saying I should put my PR for key authentication on the original repo and not yours? I'll have to go back and look at the differences to see what's involved. Or did you mean do this PR for the original repo?

mwileczka committed:
Undo revert of unrelated changes

mwileczka committed:
Bumped to 0.5.0 due to breaking changes to new/connect.

mwileczka committed:
Update revert comment
